### PR TITLE
[serve] Make HTTP proxy fault tolerant

### PR DESCRIPTION
**Pull request by edoakes**
## Why are these changes needed?

First step in making the components of serve resilient to failure. The HTTP proxy is now a reconstructable actor and fetches its configuration from the master actor on initialization.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: python/ray/actor.py [hunk 1]
```diff
@@ -545,7 +546,7 @@ def _remote(self,
             meta.language,
             meta.actor_creation_function_descriptor,
             creation_args,
-            meta.max_reconstructions,
```
Comment (simon-mo) [inline python/ray/actor.py:548]: This means we have max_reconstructions turned on by default before? I don't think that happened. What's `meta.max_reconstructions`?

**Review by simon-mo — APPROVED** [review]
Nice!

File: python/ray/actor.py (hunk 1 above)
Comment (edoakes) [reply python/ray/actor.py:548]: `meta.max_reconstructions` is just whatever was set in the decorator. This change is just allowing `max_reconstructions` to be passed using `options()`

**Comment by edoakes** [comment]
cc @stephanie-wang

**Pull request merged**